anoncreds/src/services/issuer.rs

```diff
@@ -291,24 +215,20 @@ pub fn update_revocation_registry(
 }
 
 pub fn create_credential_offer(
-    schema_id: &SchemaId,
-    cred_def: &CredentialDefinition,
+    schema_id: impl Into<SchemaId>,
+    cred_def_id: &str,
```

Comment:
Not sure why one of these is an `Into<SchemaId>` and the other a `&str`?

Reply:
Also I don't love `Into<SchemaId>` because it skips validation (can't return an Err).

Reply:
My idea was to internally use the SchemaId and for the API we use `impl Into<T>`. when we get the schema_id we call it like so:

```rust
fn test(s: impl Into<SchemaId>) -> Result<Any> {
  let s = schema_id.into().validate()?;
}
```
This way we always validate when the value comes in, but we do not require the user to cast every string into a `SchemaId`.

Reply:
It is incorrect that it is inconsistent, this should be fixed.

Reply:
It does seem more idiomatic to me for it to be a validating wrapper, ie. validation is performed when it is constructed rather than potentially being missed later. The struct(s) could have a `const fn new_unchecked(value: &str)` for skipping validation in tests or const contexts.

Reply:
Agreed. I created a `validated_new()` (did not see this comment before that) which does what your implementation of `new` would do. I think that validating with `new` is better, as it will validate by default.

I also left a comment at #25 where I propose TryInto instead of Into, which makes a lot of sense here IMO. If that seems okay with you, I can create a follow up PR where Into will be changed to TryInto.

Reply:
Implemented `new` with validation in #25 .